File: app/src/main/java/com/example/androidkotlincompose/modules/home/posts/domain/usecase/PostsInteractor.kt
package com.example.androidkotlincompose.modules.home.posts.domain.usecase

import com.example.androidkotlincompose.core.models.Resource
import com.example.androidkotlincompose.modules.home.posts.domain.models.Post
import com.example.androidkotlincompose.modules.home.posts.domain.repository.IPostsRepository

class PostsInteractor (
    private val repository: IPostsRepository
) : PostsUseCase {
    override suspend fun getPosts(): Resource<List<Post>> =
        repository.getPosts()
}

File: mobile_flutter/lib/domain/usecase/posts/posts.interactor.dart
import 'package:dartz/dartz.dart';
import 'package:mobile_flutter/core/common_models/failure.model.dart';
import 'package:mobile_flutter/data/repository/posts/posts_repository.dart';
import 'package:mobile_flutter/domain/entity/posts.entity.dart';
import 'package:mobile_flutter/domain/usecase/posts/posts.usecase.dart';

class PostsInteractor implements PostsUseCase {

  final IPostsRepository postsRepository;

  PostsInteractor({
    required this.postsRepository
  });

  @override
  Future<Either<Failure, List<PostEntity>>> getPosts() => postsRepository.getPosts();
}